### PR TITLE
REST high-level client: add update by query API

### PR DESCRIPTION
**Pull request by sohaibiftikhar**
Relates to #27205 

- [x] Requires #32679 to be merged first.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/reindex/UpdateByQueryRequest.java [hunk 1]
```diff
@@ -59,8 +64,91 @@ private UpdateByQueryRequest(SearchRequest search, boolean setDefaults) {
...
+        if (types != null) {
+            getSearchRequest().types(types);
+        }
+        return this;
+    }
+
+    /**
+     * Set the indices on which the update by query request is to be run
+     */
+    public UpdateByQueryRequest setIndices(String... indices) {
```
Comment (sohaibiftikhar) [inline server/src/main/java/org/elasticsearch/index/reindex/UpdateByQueryRequest.java]: This method is not required. Usage should be replaced by the method `indices(String... indices)`.

File: docs/java-rest/high-level/document/update-by-query.asciidoc [hunk 2]
```diff
@@ -0,0 +1,182 @@
+[[java-rest-high-document-update-by-query]]
+=== Update By Query API
+
+[[java-rest-high-document-update-by-query-request]]
+==== Update By Query Request
+
+A `UpdateByQueryRequest` can be used to update documents in an index.
+
+It requires an existing index and a target index which may or may not exist pre-request.
```
Comment (sohaibiftikhar) [inline docs/java-rest/high-level/document/update-by-query.asciidoc]: To Fix: Does not require a target index.

**Comment by nik9000** [comment]
@elasticmachine, retest this please

**Review by nik9000 — APPROVED** [review]

**Comment by alekseifedorov** [comment]
Is there any plan to merge it in 6.4.x? Thanks!

**Comment by nik9000** [comment]
I don't plan to do it, no.

**Pull request merged**